icechunk/src/change_set.rs

```diff
@@ -166,34 +276,60 @@ impl ChangeSet {
         node_id: NodeId,
         coord: ChunkIndices,
         data: Option<ChunkPayload>,
-    ) {
+    ) -> SessionResult<()> {
+        let cached_splits = self.splits.get(&node_id).expect("logic bug");
```

Comment:
let's put more info on the expect call, so we know where the error was triggered if somebody reports this error